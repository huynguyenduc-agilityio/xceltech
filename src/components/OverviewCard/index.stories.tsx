import type { Meta, StoryObj } from '@storybook/react';

// Components
import OverviewCard from '.';

const meta = {
  title: 'Components/OverviewCard',
  component: OverviewCard,
  parameters: {
    layout: 'centered',
  },

  tags: ['autodocs'],
  render: (args) => (
    <div className="flex justify-center w-[1000px] p-9 bg-blue-light">
      <OverviewCard {...args} />
    </div>
  ),
} satisfies Meta<typeof OverviewCard>;

export default meta;

type Story = StoryObj<typeof OverviewCard>;

export const Default: Story = {
  args: {
    title: 'Available Leave Days',
    children: (
      <p>
        Lorem ipsum dolor sit amet consectetur adipisicing elit. Repellat sit,
        vero ducimus voluptatibus eveniet suscipit a exercitationem blanditiis
        nisi, dicta necessitatibus quis quae placeat consequatur? Labore quas
        error iure accusantium?
      </p>
    ),
  },
};
