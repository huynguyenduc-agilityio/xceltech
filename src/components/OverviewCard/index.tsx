import { ReactNode } from 'react';

// Icons
import { KebabIcon } from '@/icons';

export interface OverviewCardProps {
  title: string;
  children: ReactNode;
}

const OverviewCard = ({ title, children }: OverviewCardProps) => {
  return (
    <div className="h-full p-10 rounded-regular bg-white">
      <div className="flex items-center justify-between mb-6">
        <span className="text-2xl font-bold">{title}</span>
        <button
          aria-label="More options"
          disabled
          className="disabled:cursor-not-allowed"
        >
          <KebabIcon />
        </button>
      </div>
      {children}
    </div>
  );
};

export default OverviewCard;
