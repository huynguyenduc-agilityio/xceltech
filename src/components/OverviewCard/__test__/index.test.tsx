import { render } from '@testing-library/react';

// Components
import OverviewCard from '..';

describe('OverviewCard component', () => {
  it('should match snapshot', () => {
    const { asFragment } = render(
      <OverviewCard
        title="Available Leave Days"
        children={
          <p>
            Lorem ipsum dolor sit amet consectetur adipisicing elit. Qui ipsum
            unde perspiciatis! Dicta officia veritatis eos expedita iste harum
            ea excepturi delectus iusto blanditiis autem voluptatibus minus
            commodi, repudiandae vero?
          </p>
        }
      />,
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
