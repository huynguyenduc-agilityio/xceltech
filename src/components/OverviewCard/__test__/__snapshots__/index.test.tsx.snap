// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OverviewCard component should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="h-[470px] p-10 rounded-regular bg-white"
  >
    <div
      class="flex items-center justify-between mb-6"
    >
      <span
        class="text-2xl font-bold"
      >
        Available Leave Days
      </span>
      <button
        aria-label="More options"
      >
        <svg
          fill="none"
          height="27"
          width="31"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M15.5 9.703c2.41 0 4.36 1.698 4.36 3.797s-1.95 3.797-4.36 3.797-4.36-1.698-4.36-3.797 1.95-3.797 4.36-3.797ZM11.14 4.22c0 2.099 1.95 3.797 4.36 3.797s4.36-1.698 4.36-3.797c0-2.1-1.95-3.797-4.36-3.797s-4.36 1.698-4.36 3.797Zm0 18.562c0 2.1 1.95 3.797 4.36 3.797s4.36-1.698 4.36-3.797-1.95-3.797-4.36-3.797-4.36 1.698-4.36 3.797Z"
            fill="#000"
          />
        </svg>
      </button>
    </div>
    <p>
      Lorem ipsum dolor sit amet consectetur adipisicing elit. Qui ipsum unde perspiciatis! Dicta officia veritatis eos expedita iste harum ea excepturi delectus iusto blanditiis autem voluptatibus minus commodi, repudiandae vero?
    </p>
  </div>
</DocumentFragment>
`;
